### ISSUE TITLE
gensms - does not alert on power loss

### ISSUE THREAD

**Issue by krisasmith** [jgyates/genmon#125]
Use the template below if you have an issue or want to report a bug. If you have a question or a feature request you can ignore the questions below.

### Expected Behavior

When utility power is lost/restored gensms sends a text

### Actual Behavior

gensms only sends an alert when the generator starts/stops. I receive email alerts for power loss.

### Steps to Reproduce (including precondition)

utility power outage

**Comment by liltux** [comment]
It appears there is not a specified message in for Utility Power Loss in the gensms.py.  I am not sure if the generator sees Utility Loss as an alarm condition.  @jgyates might be able to add that functionality.  I am not big on getting actual emails, but I use the email to txt gateway for my cellular provider.  This allows the emails to be sent to my phone as text messages, and I get every notification sent to my cell phone.  Just an example, verizon account would be - (ten digit phone number)@vtext.com.  Just for information.

**Comment by jgyates** [comment]
gensms.py relies on mynotify.py (as do a couple of the other programs like pushover). mynotify.py notifies on changes in the base state of the generator. Currently there are eight states:

     READY, EXERCISING, RUNNING, RUNNING-MANUAL, ALARM, SERVICEDUE, MANUAL, OFF

You should receive notices when the generator starts and stops. During a power outage the stop time will be slightly later (1 - 5 min) later than the time that utility power is restored due to the generator is in cool down mode.

Adding these notices at power loss and power restore events is possible, but there are ramifications since mynotify.py is based on generator **state**, not **events**, i.e. in the current implementation you will get a notice that one state ends when another begins, but that will not hold true if we add utility loss as you can have multiple events occurring at the same time. 

an alternate solution is to add some code to gensms that will monitor for power outages and send text, this is likely the better solution as it keeps the mynotify.py state model and will add notices for power outage/restore events.

Long story short, I will add this to the list.

**Comment by krisasmith** [comment]
I may have a unique situation. I see a few times a month a power outage that lasts 10 seconds or so. I set the startup delay to 60 seconds so the generator does not start. I get emails telling me but no sms. Thanks for adding it to your to do list.

**Comment by jgyates** [comment]
Hi Kris,

Check out the latest and give it a try. I updated gensms, genpushover and gensyslog to have the same notification of utility power state change. Let me know if you have any issues. 

Thanks
Jason

**Comment by jgyates** [comment]
Hi Kris, I am closing this issue. If you have any problems once you get to this let me know.
